Nodejs/Product/TestAdapter/TestFilesUpdateWatcher.cs

```diff
@@ -43,84 +36,47 @@ public bool AddDirectoryWatch(string path)
         {
             ValidateArg.NotNull(path, "path");
 
-            if (!string.IsNullOrEmpty(path))
+            if (!this.watchedFiles.ContainsKey(path) && ErrorHandler.Succeeded(this.fileWatcher.AdviseDirChange(path, VSConstants.S_OK, this, out uint cookie)))
             {
-                if (!_fileWatchers.ContainsKey(path) && Directory.Exists(path))
-                {
-                    var watcher = new FileSystemWatcher(path);
-                    _fileWatchers[path] = watcher;
-
-                    watcher.IncludeSubdirectories = true;
-                    watcher.Changed += OnChanged;
-                    watcher.Renamed += OnRenamed;
-                    watcher.EnableRaisingEvents = true;
-                    return true;
-                }
+                this.watchedFiles.Add(path, cookie);
+                return true;
             }
             return false;
         }
 
-        public void RemoveWatch(string path)
+        public bool RemoveWatch(string path)
         {
             ValidateArg.NotNull(path, "path");
 
-            if (!string.IsNullOrEmpty(path))
+            if (this.watchedFiles.TryGetValue(path, out uint cookie))
             {
-                FileSystemWatcher watcher;
-
-                if (_fileWatchers.TryGetValue(path, out watcher))
+                this.watchedFiles.Remove(path);
+                if (Path.HasExtension(path))
```

Comment:
The presence or absence of an extension is a very error prone way to determine if something is a directory or not. Please make this code more determinisitc.

Reply:
Considering we're only dealing with `.js` or `.ts` files this works for this case. We don't want to hit the disk when we stop watching a file/folder.

I'll add a comment though.

Reply:
Put the burden of getting this right on the user.